saleor/payment/utils.py

```diff
@@ -189,12 +212,91 @@ def create_payment_information(
         reuse_source=store_source,
         data=additional_data or {},
         graphql_customer_id=graphql_customer_id,
+        refund_data=refund_data,
         _resolve_lines=lambda: create_payment_lines_information(
             payment, manager or get_plugins_manager()
         ),
     )
 
 
+RefundLines = Iterator[Tuple[Any, OrderLine]]
+
+
+def _prepare_refund_lines(
+    fulfillments: QuerySet[Fulfillment],
+    order_lines_to_refund: List[OrderLineData],
+    fulfillment_lines_to_refund: List[FulfillmentLineData],
+) -> Iterator[Tuple[int, int]]:
+    previous_refund_lines = chain.from_iterable(
+        (
+            (p_variant_id, line.quantity)
+            for line in f.lines.filter(order_line__variant_id__isnull=False)
+            if (p_variant_id := line.order_line.variant_id)
+        )
+        for f in fulfillments.all()
+    )
+
+    current_order_refund_lines = (
+        (o_variant_id, line.quantity)
+        for line in order_lines_to_refund
+        if (o_variant_id := line.line.variant_id)
```

Comment:
You can use `variant` field from `OrderLineData`.
```suggestion
        if (o_variant_id := line.variant.id)
```

Reply:
After changing from `line.line.variant_id` to `line.variant.id`, refund stopped working. If I understand it correctly, mutation `FulfillmentRefundAndReturnProductBase` when cleaning order lines, does not set `OrderLineData.variant`. Is this a correct behaviour?